system/jlib/jencrypt.cpp

```diff
@@ -1891,6 +1910,56 @@ MemoryBuffer &aesEncrypt(const void *key, size_t keylen, const void *plaintext,
     }
 }
 
+extern jlib_decl size_t aesEncryptInPlace(const void *key, size_t keylen, void *buffer, size_t inlen, size_t buflen)
+{
+    if (!buffer || !inlen)
+        return 0;
+    byte *data = (byte *) buffer;
+    EVP_CIPHER_CTX *ctx = EVP_CIPHER_CTX_new();
+    if (!ctx)
+        encryptError("Failed to create context");
```

Comment:
if we throw here, is it ok to call EVP_CIPHER_CTX_free(ctx) without a valid ctx ?  Probably.